lib/matplotlib/tests/test_legend.py

```diff
@@ -1375,3 +1377,16 @@ def test_legend_handle_label_mismatch_no_len():
                         labels=iter(["pl1", "pl2", "pl3"]))
     assert len(legend.legend_handles) == 2
     assert len(legend.get_texts()) == 2
+
+
+def test_legend_nolabels_warning():
+    plt.plot([1, 2, 3])
+    with pytest.raises(UserWarning):
```

Comment:
I like to be sure to catch the correct warning:
```suggestion
    with pytest.raises(UserWarning, match="No artists with labels found"):
```